internals/daemon/api_notices.go

```diff
@@ -42,6 +45,38 @@ type addedNotice struct {
 func v1GetNotices(c *Command, r *http.Request, _ *UserState) Response {
 	query := r.URL.Query()
 
+	requestUID, err := uidFromRequest(r)
+	if err != nil {
+		return statusForbidden("cannot determine UID of request, so cannot retrieve notices")
+	}
+	// By default, return notices with the request UID and public notices.
+	userID := &requestUID
+	daemonUID := uint32(sysGetuid())
+
+	if len(query["user-id"]) > 0 {
+		if !isAdmin(requestUID, daemonUID) {
+			return statusForbidden(`only admins may use the "user-id" filter`)
+		}
+		userID, err = sanitizeUserIDFilter(query["user-id"])
+		if err != nil {
+			return statusBadRequest(`invalid "user-id" filter: %v`, err)
+		}
+	}
+
+	if len(query["select"]) > 0 {
+		if !isAdmin(requestUID, daemonUID) {
+			return statusForbidden(`only admins may use the "select" filter`)
+		}
+		if query["user-id"] != nil {
+			return statusBadRequest(`cannot use both "select" and "user-id" parameters`)
+		}
+		if err = sanitizeSelectFilter(query["select"]); err != nil {
+			return statusBadRequest(`invalid "select" filter: %v`)
+		}
+		// Clear the userID filter so all notices will be returned
+		userID = nil
```

Comment:
Nice comment, and I like how you default to the request UID above and clear it here.